core-rust/core-api-server/src/jni/mod.rs

```diff
@@ -142,10 +145,17 @@ extern "system" fn Java_com_radixdlt_api_CoreApiServer_start(
 
         let state = jni_core_api_server.state.clone();
         let runtime = &jni_core_api_server.runtime;
+        let metric_registry = jni_core_api_server.metric_registry.clone();
```

Comment:
hmm, could we call a `CoreApiMetrics::new(&registry)` here and pass that instance down? Would it be less cloning or `Arc`ing? Would it be more Rust-y?

_(just asking, I am the Rust noob here :p)_

Reply:
Yes we can do that, there is no "Rust downside", i.e. clone-wise (it's not on the hot-path).
I just want to keep the pattern of setup metrics inside component that is using it (no point having `CoreApiMetrics` here since it's not shared with anything else). Also this implies `create_server` is a "leaf" in the "Metrics tree" (which might not be true later).

If you're wondering why not just passing `&jni_core_api_server.metric_registry` it's because that means capturing whole `jni_core_api_server` which is not `Send`.